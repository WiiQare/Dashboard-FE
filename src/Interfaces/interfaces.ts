export type UserType = {
    id: string;
    name?: string | null | undefined;
    email?: string | null | undefined;
    image?: string | null | undefined;
    data: {
        userId: string;
        names: string;
        type: string;
    };
};
