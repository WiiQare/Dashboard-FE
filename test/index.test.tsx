// Home.test.tsx
import React from 'react';
import { render } from '@testing-library/react';
import Home from '@/pages';
import { UserType } from '@/Interfaces/interfaces';
import { SessionProvider } from 'next-auth/react';

test('renders Home component', () => {
  const { container } = render(
    <SessionProvider
      session={{
        user: { data: { userId: 'random123' } } as UserType,
        expires: '',
      }}
    >
      <Home />)
    </SessionProvider>,
  );
  expect(container).toMatchSnapshot();
});
