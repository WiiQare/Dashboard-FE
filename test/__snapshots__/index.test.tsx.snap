// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component 1`] = `
<div>
  <div
    class="transition-c-0.5  overflow-y-auto p-4 h-full w-full sm:px-4"
    id="main"
  >
    <div
      class="grid grid-cols-1 gap-4 mb-4"
    >
      <div
        class="w-full"
      >
        <div
          class="grid cards grid-cols-1 gap-1 md:grid-cols-3 md:gap-4 xl:grid-cols-5 2xl:gap-7.5 p-[23px]"
        >
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3 w-3/4 md:w-2/3 lg:w-1/2 2xl:w-3/4"
              >
                <div
                  class="w-1/2 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-2/3 h-4 bg-gray-300 rounded-md"
                />
                <div
                  class="w-full h-4 bg-gray-300 rounded-md"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="p-4"
        >
          <div
            class="flex flex-col md:h-[23rem] lg:flex-row gap-4 transition-c-0.5 pr-2"
          >
            <div
              class="flex flex-grow dark:bg-[#182644] border-[#180a0a07] border-[0.2px] shadow-md rounded-md bg-white lg:w-1/2"
            >
              <div
                class="p-chart"
                data-pc-name="chart"
                data-pc-section="root"
                style="width: 100%; height: 100%; font-family: inherit; padding: 0.5rem;"
              >
                <canvas
                  data-pc-section="canvas"
                  height="100%"
                  role="img"
                  width="100%"
                />
              </div>
            </div>
            <div
              class="flex flex-grow dark:bg-[#182644] border-[#180a0a07] border-[0.2px] cheatcode shadow-md rounded-md bg-white lg:w-1/2"
            />
          </div>
          <div
            class="flex mt-5 pr-5 flex-grow dark:bg-[#182644] border-[#180a0a07] border-[0.2px] shadow-md rounded-md h-[28rem] bg-white"
          >
            <div
              class="w-[95%]"
            >
              <div
                class="bar-chart MuiBox-root css-1amhwt3"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  )
</div>
`;
